src/electron/routing_service.ts

```diff
@@ -65,75 +63,79 @@ enum RoutingServiceStatusCode {
 //  - Linux: systemctl start|stop outline_proxy_controller.service
 //  - Windows: net start|stop OutlineService
 export class RoutingDaemon {
-  private socket: Socket|undefined;
+  private socket: Socket | undefined;
 
   private stopping = false;
 
   private fulfillDisconnect!: () => void;
 
-  private disconnected = new Promise<void>((F) => {
+  private disconnected = new Promise<void>(F => {
     this.fulfillDisconnect = F;
   });
 
   private networkChangeListener?: (status: TunnelStatus) => void;
 
   constructor(private proxyAddress: string, private isAutoConnect: boolean) {}
 
+  /**
+   * Make sure the routing daemon is installed in the system.
+   * A SystemConfigurationException will be thrown if installation failed.
+   */
+  public async ensureDaemonInstalled() {
+    if (await this.isDaemonInstalled()) {
+      return;
+    }
+    if (isWindows) {
+      throw new errors.SystemConfigurationException("please rerun Outline installer in Windows");
```

Comment:
We need to surface a localized message to the user, though this is not really the right place for UI.
Return an error that the UI can identify and display the right message to the user.